src/ConfigParser/ConfigParser.js

```diff
@@ -168,6 +168,23 @@ ConfigParser.prototype = {
     getPlatformPreference: function (name, platform) {
         return findElementAttributeValue(name, this.doc.findall('./platform[@name="' + platform + '"]/preference'));
     },
+    setPlatformPreference: function (name, platform, value) {
+        const platformEl = this.doc.find('./platform[@name="' + platform + '"]');
+        if (!platformEl) {
+            throw new CordovaError('platform does not exist (received platform: ' + platform + ')');
```

Comment:
Wouldn't it be nicer if we created the `platform` element here instead of throwing?

Reply:
I guess it's much simpler to just throw. A new platform won't work automatically by just add a new section in the config.